prow/crier/reporters/resultstore/reporter.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resultstore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/crier/reporters/gcs/util"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Reporter reports Prow results to ResultStore and satisfies the
+// crier.reportClient interface.
+type Reporter struct {
+	cfg config.Getter
+}
+
+// New returns a new Reporter.
+func New(cfg config.Getter) *Reporter {
+	return &Reporter{
+		cfg: cfg,
+	}
+}
+
+// GetName returns the name of this reporter.
+func (r *Reporter) GetName() string {
+	return "resultstoreReporter"
+}
+
+// ShouldReport returns whether results should be reported for this
+// job at this time.
+func (r *Reporter) ShouldReport(ctx context.Context, log *logrus.Entry, pj *v1.ProwJob) bool {
+	if !pj.Spec.Report {
+		return false
+	}
+
+	// ResultStore requires files stored in GCS; use the GCS reporter's
+	// heurisitic to determine the GCS configuration and bucket.
+	if b, _, _ := util.GetJobDestination(r.cfg, pj); !isGCSBucket(b) {
+		return false
+	}
+
+	if pj.Status.State == v1.TriggeredState || pj.Status.State == v1.PendingState {
+		log.Info("job not finished")
+		return false
+	}
+
+	// TODO: Per the design, determine ProjectID, and whether prow
+	// has resultstore write permissions. If not, log an error and
+	// return false. If this is expensive (e.g. I/O), it should not
+	// be checked here, but an error returned by Report.
+
+	return true
+}
+
+// Report reports results for this job to resultstore.
+func (r *Reporter) Report(ctx context.Context, log *logrus.Entry, pj *v1.ProwJob) ([]*v1.ProwJob, *reconcile.Result, error) {
+	return nil, nil, fmt.Errorf("not yet implemented")
+}
+
+func isGCSBucket(bucket string) bool {
```

Comment:
I'm surprised we don't have a similar function already. My guess is because we have https://github.com/kubernetes/test-infra/blob/3a69ac1572cf4a7878ced659de235310bee28d2b/prow/io/opener.go#L100 (Opener) for dealing with paths (not sure if any of that is relevant, because IIRC this reporter will not upload files to GCS itself at all but merely check that some files there exist and if so, report them to ResultStore).

Reply:
Sorry, not sure if there's something actionable here?

Reply:
Nope, just thinking out loud from my end.

Well, also the `isGCSBucket` function probably belongs outside of reporter.go in some other GCS-related area, but that is a very minor concern.

Reply:
Added a TODO. Moving this to util.go would allow a simpler impl. I did notice some other bits related to bucket identification (e.g. in config.go) but that doesn't seem the right place.